src/function/execute.rs

```diff
@@ -105,6 +105,7 @@ where
             // outputs and update the tracked struct IDs for seeding the next revision.
             self.diff_outputs(zalsa, database_key_index, old_memo, &mut revisions);
         }
+        revisions.tracked_struct_ids.shrink_to_fit();
```

Comment:
Should we move this call to `Memo::new`? It would then also apply for memos participating in cycles